### PR TITLE
IndexIterator no longer nexts prior to seeking, updated docs

### PR DESCRIPTION
**Pull request by apmoriarty**

### CODE REVIEW

**Review by FineAndDandy — DISMISSED** [review]

File: warehouse/query-core/src/main/java/datawave/query/iterator/logic/IndexIterator.java [hunk 1]
```diff
@@ -137,8 +137,7 @@ private IndexIterator(Text field, Text value, SortedKeyValueIterator<Key,Value>
         
         this.source = source;
         // wrap the source with a limit
-        limitedSource = new LimitedSortedKeyValueIterator(source);
-        this.timeFilter = timeFilter;
+        this.limitedSource = new LimitedSortedKeyValueIterator(source);
```
Comment (drewfarris) [inline warehouse/query-core/src/main/java/datawave/query/iterator/logic/IndexIterator.java:140]: Consider making the `limitedSource` field `final`

File: warehouse/query-core/src/main/java/datawave/query/iterator/logic/IndexIterator.java [hunk 2]
```diff
@@ -137,8 +137,7 @@ private IndexIterator(Text field, Text value, SortedKeyValueIterator<Key,Value>
         
         this.source = source;
```
Comment (drewfarris) [inline warehouse/query-core/src/main/java/datawave/query/iterator/logic/IndexIterator.java:138]: Consider making the `source` field `final`

File: warehouse/query-core/src/main/java/datawave/query/iterator/logic/IndexIterator.java (hunk 1 above)
Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/query/iterator/logic/IndexIterator.java:140]: done

File: warehouse/query-core/src/main/java/datawave/query/iterator/logic/IndexIterator.java (hunk 2 above)
Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/query/iterator/logic/IndexIterator.java:138]: done

**Review by FineAndDandy — APPROVED** [review]

**Review by ivakegg — APPROVED** [review]

**Review by FineAndDandy — APPROVED** [review]

**Pull request merged**